apps/dashboard/app/models/workflow.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class Workflow
+  include ActiveModel::Model
+
+  class << self
+    def workflows_file(project_dir)
+      workflows_dir = Pathname.new("#{project_dir}/.ondemand/workflows")
+      # TODO: Later use <workflow_id>.yml to get list of all workflows
+      file = workflows_dir.join("workflow.yml")
+
+      FileUtils.mkdir_p(workflows_dir)
+      FileUtils.touch(workflows_dir.join("workflow.yml")) unless file.exist?
+
+      file
+    end
+
+    def all(project_id)
+      project = Project.find(project_id)
+      project_dir = project.directory
+      f = File.read(workflows_file(project_dir))
+      YAML.safe_load(f).to_h.map do |id, workflow_name|
+        Workflow.new({ id: id, name: workflow_name })
+      end
+    rescue StandardError, Exception => e
+      Rails.logger.warn("cannot read #{project_dir}/.ondemand/workflows/workflow.yml due to error #{e}")
+      {}
+    end
```

Comment:
This is still looking for 1 single workflow file - is that expected? Shouldn't this be capable of having multiple workflows?

Reply:
No this will give out list of workflows.

`workflow.xml` will keep list of all `<workflow-id , workflow-name>` just like the lookup file in projects. Later once we have <workflow-id>.yml files in place then we can remove this logic if we like.

Reply:
OK i guess, but it seems like you'd need another hash under id. Like it can't just be the id is the key and the value is the name. Seems like the value would actually be a hash of more key/value pairs.